src/testsuite/vec_int64_dummy.c

```diff
@@ -486,4 +552,138 @@ __test_cmpleud (vui64_t a, vui64_t b)
 {
   return vec_cmple (a, b);
 }
+
+vui64_t
+__test_vmuludm (vui64_t vra, vui64_t vrb)
+{
+  vui64_t s32 = (vui64_t) { 32, 32 };
+  vui64_t z = (vui64_t) { 0, 0 };
+  vui64_t t4;
+  vui64_t t2, t3;
+  vui32_t t1;
+
+  t1 = (vui32_t)vec_vrld (vrb, s32);
+  t2 = vec_mulouw ((vui32_t)vra, (vui32_t)vrb);
```

Comment:
Perhaps a fundamental misunderstanding on my part, but why not use the type-generic APIs, like `vec_mulo` here, whenever possible?

Reply:
This is from __test_vmuludm_PWR7 ?
two reasons:
1) This optimization requires a mix of word and double word operations and I need to keep them all straight. So vec_mulouw is the operation I need and I can not mess up by using the wrong cast (which I could with vec_mulo).
2) vmulouw is not defined for _ARCH_PWR7 and the generic vec_mulo (vector int) is not defined until AT11. vec_mulouw is actually defined in vec_int32_ppc,h and handles all the special (wierd) cases. If you used the SDK, mouse over would have told you this ;)